### PR TITLE
fix: ReadOnlyDirectory should provide the largest abs(cycle) when cycle is unspecified, not the largest cycle (v4).

### PR DESCRIPTION
**Pull request by jpivarski**
This is the "v4" version of #715.

Like the other in-flight PRs, it will follow this procedure:

  1. #720 gets approved and merged into `main-v4`.
  2. This PR updates to `main-v4`.
  3. The tests re-run and pass (again).
  4. This PR gets approved and merged into `main-v4`.

### CODE REVIEW

File: src/uproot/reading.py [hunk 1]
```diff
@@ -2047,7 +2047,7 @@ def key(self, where):
                 return key
             elif cycle is None and last is None:
                 last = key
-            elif cycle is None and last.fCycle < key.fCycle:
+            elif cycle is None and abs(last.fCycle) < abs(key.fCycle):
```
Comment (agoose77) [inline src/uproot/reading.py:2050]:
```suggestion
            # Follow ROOT's behaviour in comparing negative fCycle values
            elif cycle is None and abs(last.fCycle) < abs(key.fCycle):
```

Comment (agoose77) [reply src/uproot/reading.py:2050]: Just adding a comment for future ease of understanding.

(yes, we can git-blame, but this is low-cost high-readability).

**Comment by jpivarski** [comment]
I'll take that as an approval of both #721 and #715 and enable auto-merge.

**Comment by agoose77** [comment]
Kicking CI

**Pull request merged**